src/KurrentDB.SecondaryIndexing.Tests/IntegrationTests/IndexingEnabledTests.cs

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) Kurrent, Inc and/or licensed to Kurrent, Inc under one or more agreements.
+// Kurrent, Inc licenses this file to you under the Kurrent License v1 (see LICENSE.md).
+
+using KurrentDB.Core.Data;
+using KurrentDB.SecondaryIndexing.Indexes.Category;
+using KurrentDB.SecondaryIndexing.Indexes.Default;
+using KurrentDB.SecondaryIndexing.Indexes.EventType;
+using KurrentDB.SecondaryIndexing.Tests.Fixtures;
+using KurrentDB.SecondaryIndexing.Tests.Generators;
+using Xunit.Abstractions;
+
+namespace KurrentDB.SecondaryIndexing.Tests.IntegrationTests;
+
+[Trait("Category", "Integration")]
+[Collection("SecondaryIndexingPluginEnabled")]
+public class IndexingEnabledTests(
+	SecondaryIndexingEnabledFixture fixture,
+	ITestOutputHelper output
+) : SecondaryIndexingTestBase(fixture, output) {
+	private readonly MessageGenerator _messageGenerator = new();
+
+	private readonly LoadTestPartitionConfig _config = new(
+		PartitionId: 1,
+		StartCategoryIndex: 0,
+		CategoriesCount: 5,
+		MaxStreamsPerCategory: 100,
+		MessageTypesCount: 10,
+		MessageSize: 10,
+		MaxBatchSize: 2,
+		TotalMessagesCount: 1000
+	);
+
+	public override async Task BeforeAll() {
+		await foreach (var batch in _messageGenerator.GenerateBatches(_config)) {
+			var messages = batch.Messages.Select(m => m.ToEventData()).ToArray();
+			await fixture.AppendToStream(batch.StreamName, messages);
+			AppendedBatches.Add(batch);
+		}
+	}
+
+	[Fact]
+	public Task ReadsAllEventsFromDefaultIndex() =>
+		ValidateRead(DefaultIndex.Name, AppendedBatches.ToDefaultIndexResolvedEvents());
+
+	[Fact]
+	public async Task ReadsAllEventsFromCategoryIndex() {
+		foreach (var category in Categories) {
+			var expectedEvents = AppendedBatches.ToCategoryIndexResolvedEvents(category);
+			await ValidateRead(CategoryIndex.Name(category), expectedEvents);
+		}
+	}
+
+	[Fact]
+	public async Task ReadsAllEventsFromEventTypeIndex() {
+		foreach (var eventType in EventTypes) {
+			var expectedEvents = AppendedBatches.ToEventTypeIndexResolvedEvents(eventType);
+			await ValidateRead(EventTypeIndex.Name(eventType), expectedEvents);
+		}
+	}
+
+	[Fact]
+	public Task SubscriptionReturnsAllEventsFromDefaultIndex() =>
+		ValidateSubscription(DefaultIndex.Name, AppendedBatches.ToDefaultIndexResolvedEvents());
+
+	[Fact]
+	public async Task SubscriptionReturnsAllEventsFromCategoryIndex() {
+		foreach (var category in Categories) {
+			var expectedEvents = AppendedBatches.ToCategoryIndexResolvedEvents(category);
+			await ValidateSubscription(CategoryIndex.Name(category), expectedEvents);
+		}
+	}
+
+	[Fact]
+	public async Task SubscriptionReturnsAllEventsFromEventTypeIndex() {
+		foreach (var eventType in EventTypes) {
+			var expectedEvents = AppendedBatches.ToEventTypeIndexResolvedEvents(eventType);
+			await ValidateSubscription(EventTypeIndex.Name(eventType), expectedEvents);
+		}
```

Comment:
it might be worth adding a test when the subscription reaches the end of the stream and goes live too:ie append a few new events and see if they are received by the subscription